database/store.go

```diff
@@ -90,42 +81,43 @@ func (s store) AddEvent(in *pb.AddEventRequest) (string, error) {
 	return "Event correctly added!", nil
 }
 
-func (s store) AddTeam(in *pb.AddTeamRequest) (string, error) {
+func (s *store) AddTeam(in *pb.AddTeamRequest) (string, error) {
 	s.m.Lock()
 	defer s.m.Unlock()
 
 	now := time.Now()
 	nowString := now.Format(timeFormat)
-	_, err := s.db.Exec(ADD_TEAM_QUERY, in.Id, in.EventTag, in.Email, in.Name, in.Password, nowString, nowString, "[]")
+
+	var eventId int
+	if err := s.db.QueryRow(QUERY_EVENT_ID, in.EventTag).Scan(&eventId); err != nil {
+		return "", err
+	}
+
+	_, err := s.db.Exec(ADD_TEAM_QUERY, in.Id, eventId, in.Email, in.Name, in.Password, nowString, nowString, "[]")
 	if err != nil {
 		return "", err
 	}
 	return "Team correctly added!", nil
 }
 
-func (s store) GetEvents() ([]model.Event, error) {
+func (s *store) GetEvents() ([]model.Event, error) {
 
 	s.m.Lock()
 	defer s.m.Unlock()
 
 	rows, err := s.db.Query(QUERY_EVENT_TABLE)
 	if err != nil {
-		return []model.Event{}, err
+		return nil, err
 	}
 	var events []model.Event
 	for rows.Next() {
 
-		// todo  : optimize this one, looks ugly
-		var tag string
-		var name string
-		var frontends string
-		var exercises string
-		var available uint
-		var capacity uint
-		var startedAt string
-		var expectedFinishTime string
-		var finishedAt string
-		rows.Scan(&tag, &name, &available, &capacity, &frontends, &exercises, &startedAt, &expectedFinishTime, &finishedAt)
+		var tag, name, frontends, exercises, startedAt, expectedFinishTime, finishedAt string
+		var id, available, capacity  uint
+		err := rows.Scan(&id, &tag, &name, &available, &capacity, &frontends, &exercises, &startedAt, &expectedFinishTime, &finishedAt)
```

Comment:
There should be better approach for handling this kind of multiple variables, in rows.

This blog post could might be helpful for better handling this kind of cases:  https://www.alexedwards.net/blog/using-postgresql-jsonb